### PR TITLE
fix rw rootimg statelite for sles12.2

### PR DESCRIPTION
**Pull request by bybai**
Fix issue #2599 the entire statelite image is rw in sles12sp2 x86_64 

Use liteimg change rw to ro for `/` dir in rootimg/etc/fstab

### CODE REVIEW

**Comment by immarvin** [comment]
agree to merge

File: xCAT-server/lib/xcat/plugins/statelite.pm [hunk 1]
```diff
@@ -181,6 +181,8 @@ sub process_request {
     system("mkdir -p $rootimg_dir/xcatpost");
     system("cp -r $installroot/postscripts/* $rootimg_dir/xcatpost/");
 
+    # update rw to ro for sles
```
Comment (daniceexi) [inline xCAT-server/lib/xcat/plugins/statelite.pm]: If this change only for sles, we should show it in the name of subroutine and also make this subroutine only be called for sles.

File: xCAT-server/lib/xcat/plugins/statelite.pm [hunk 2]
```diff
@@ -968,5 +971,33 @@ sub liteItem {
     }
 }
 
+=head3
+    updateFstab
```
Comment (daniceexi) [inline xCAT-server/lib/xcat/plugins/statelite.pm]: we may add description for the new subroutine.

File: xCAT-server/lib/xcat/plugins/statelite.pm [hunk 3]
```diff
@@ -968,5 +971,33 @@ sub liteItem {
...
+=cut
+
```
Comment (daniceexi) [inline xCAT-server/lib/xcat/plugins/statelite.pm]: I am a little surprise for the root cause of this issue. I was thinking that the root fs won't be remounted after the mount in the initrd.

From your change, looks like the mount option of rootfs is controlled by the /etc/fstab. That means the rootfs for nfs-based statelite is remounted after chroot base on the entry in the /etc/fstab, am I right?

BTW, is the statelite for sles12 using dracut or not?

Comment (bybai) [inline xCAT-server/lib/xcat/plugins/statelite.pm]: Hi @daniceexi ,
The statelite for sles12 is using dracut.

In initrd, the sysroot is ro mounted, after cleanup and switchroot, it use systemd remount, it is based on /etc/fstab.

File: xCAT-server/lib/xcat/plugins/statelite.pm (hunk 1 above)
Comment (bybai) [inline xCAT-server/lib/xcat/plugins/statelite.pm]: Hi @daniceexi , In redhat, /etc/fstab is different from sles, there is no the same line, so the function will no run in redhat.

**Pull request merged**